File: exercism/csharp/hamming/Hamming.cs
using System.Linq;

namespace Exercism
{
    public static class Hamming
    {
        public static int Compute(string xs, string ys)
        {
            return (from result in xs.Zip(ys, (x, y) => x == y)
                    where result == false
                    select result).Count();
        }
    }
}

File: exercism/cpp/hamming/hamming.cpp
#include "hamming.h"

std::size_t hamming::compute(const std::string &xs, const std::string &ys) {

  if (xs.length() != ys.length())
    throw std::domain_error("String lengths don't match");

  std::size_t count = 0;
  for (int i = 0; i < xs.length(); i++) {
    if (xs[i] != ys[i])
      count++;
  }
  return count;
}
